services/inc/LinkLossService.h

```diff
@@ -0,0 +1,126 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_LINK_LOSS_SERVICE_H
+#define BLE_LINK_LOSS_SERVICE_H
+
+#ifdef BLE_FEATURE_GATT_SERVER
+
+#include "ble/BLE.h"
+#include "ble/Gap.h"
+#include "events/EventQueue.h"
+#include "ble/gap/ChainableGapEventHandler.h"
+
+#include <chrono>
+
+/**
+ * BLE Link Loss Service
+ *
+ * @par purpose
+ *
+ * @par usage
+ *
+ * @note The specification for the link loss service can be found here:
+ * https://www.bluetooth.com/specifications/gatt
+ *
+ * @attention The user should not instantiate more than a single link loss service
+ */
+class LinkLossService : private ble::Gap::EventHandler {
+public:
+    enum class AlertLevel : uint8_t {
+        NO_ALERT    = 0,
+        MILD_ALERT  = 1,
+        HIGH_ALERT  = 2
+    };
+
+    struct EventHandler {
+        virtual void on_alert_requested(AlertLevel) { }
+        virtual void on_alert_end() { }
+    };
+
+    /**
+     * Construct and initialize a link loss service.
+     *
+     * @param ble
+     * @param event_queue
+     * @param chainable_gap_event_handler
+     */
+    LinkLossService(BLE &ble, events::EventQueue &event_queue, ChainableGapEventHandler &chainable_gap_event_handler);
+
+    /**
+     *
+     */
+    ~LinkLossService();
```

Comment:
This class is not copyable or movable, please delete the copy constructor and copy assignment operator: 

```c++
LinkLossService(const LinkLossService&) = delete;
LinkLossService &operator=(const LinkLossService&) = delete;
```